driver/src/main/com/mongodb/client/MongoCollection.java

```diff
@@ -163,7 +163,7 @@
      * @return a non-null list of distinct values
      * @mongodb.driver.manual reference/command/distinct/ Distinct
      */
-    List<Object> distinct(String fieldName, Object filter, DistinctOptions distinctOptions);
+    MongoIterable<Object> distinct(String fieldName, Object filter, DistinctOptions distinctOptions);
```

Comment:
I cant see how this can be accomplished at the moment as we have to embed the values in a BsonDocument and then decode.  With a `decoder<T>` we lose the ability to get those values out of the decoded item :(